hadoop-common-project/hadoop-common/src/main/java/io/hops/security/ServiceJWTManager.java

```diff
@@ -0,0 +1,592 @@
+    this.tokenRenewer = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
+        .setNameFormat("JWT renewer thread")
+        .setDaemon(true)
+        .build());
+    waiter = new CountDownLatch(1);
+  }
+  
+  public void setHTTPClient(CloseableHttpClient client) {
+    this.httpClient = client;
+  }
+  
+  @Override
+  protected void serviceInit(Configuration conf) throws Exception {
+    host = new URL(conf.get(CommonConfigurationKeysPublic.HOPS_HOPSWORKS_HOST_KEY,
+        "https://127.0.0.1"));
+    sslConf = new Configuration(false);
+    sslConf.addResource(conf.get(SSLFactory.SSL_SERVER_CONF_KEY, "ssl-server.xml"));
+    loadMasterJWT();
+    loadRenewalJWTs();
+    serviceJWTValidityPeriodSeconds = conf.getTimeDuration(
+        CommonConfigurationKeysPublic.JWT_MANAGER_MASTER_TOKEN_VALIDITY_PERIOD,
+        CommonConfigurationKeysPublic.DEFAULT_JWT_MANAGER_MASTER_TOKEN_VALIDITY_PERIOD,
+        TimeUnit.SECONDS);
+    if (serviceJWTValidityPeriodSeconds == 0) {
+      serviceJWTValidityPeriodSeconds = 30L;
+    }
+    String serviceJWTRenewPathConf = conf.get(CommonConfigurationKeysPublic.JWT_MANAGER_SERVICE_RENEW_PATH,
+        CommonConfigurationKeysPublic.DEFAULT_JWT_MANAGER_SERVICE_RENEW_PATH);
+    serviceJWTRenewPath = new URL(host, serviceJWTRenewPathConf);
+    String serviceJWTInvalidatePathConf = conf.get(CommonConfigurationKeysPublic.JWT_MANAGER_SERVICE_INVALIDATE_PATH,
+        CommonConfigurationKeysPublic.DEFAULT_JWT_MANAGER_SERVICE_INVALIDATE_PATH);
+    if (!serviceJWTInvalidatePathConf.endsWith("/")) {
+      serviceJWTInvalidatePathConf = serviceJWTRenewPathConf + "/";
+    }
+    serviceJWTInvalidatePath = new URL(host, serviceJWTInvalidatePathConf);
+    super.serviceInit(conf);
+  }
+  
+  private void createLockFileIfMissing() throws IOException {
+    Path lockFile = getLockFilePath();
+    if (!lockFile.toFile().exists()) {
+      FileChannel fc = FileChannel.open(lockFile, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
+      FileLock fl = fc.tryLock();
+      if (fl == null) {
+        return;
+      }
+      try {
+        Files.setPosixFilePermissions(lockFile, LOCK_FILE_PERMISSIONS);
+      } finally {
+        fl.release();
+      }
+    } else {
+      Files.setPosixFilePermissions(lockFile, LOCK_FILE_PERMISSIONS);
+    }
+  }
+  
+  private Path getLockFilePath() {
+    String tmp = System.getProperty("java.io.tmpdir");
+    return Paths.get(tmp, LOCK_FILE);
+  }
+  
+  @Override
+  protected void serviceStart() throws Exception {
+    createLockFileIfMissing();
+    tokenRenewer.execute(new TokenRenewer());
+    super.serviceStart();
+  }
+  
+  @Override
+  protected void serviceStop() throws Exception {
+    try {
+      tokenRenewer.shutdown();
+      if (!tokenRenewer.awaitTermination(100L, TimeUnit.MILLISECONDS)) {
+        tokenRenewer.shutdownNow();
+      }
+    } catch (InterruptedException ex) {
+      tokenRenewer.shutdownNow();
```

Comment:
Shouldn't we call Thread.interupt after that to not swallow the interupt.

Reply:
Same as above